src/models/pull-request.ts

```diff
@@ -0,0 +1,172 @@
+import { githubGraphqlClient } from '../utils/env';
+import { singlePullRequest } from '../utils/graphql-queries';
+import { DateTime, Duration } from 'luxon';
+import { memoize } from '../utils/decorators';
+import debug, { Debugger } from '../utils/debug';
+export async function loadPullRequest(number: number): Promise<PullRequest> {
+  let graphql = githubGraphqlClient();
+  let data: any = await graphql(singlePullRequest(number));
+  return new PullRequest(data.repository.pullRequest as GraphQLPRData);
+}
+
+function findLast<T>(arr: Array<T>, predicate: (item: T) => boolean): T | null {
+  for (let i = arr.length - 1; i >= 0; i--) {
+    if (predicate(arr[i])) {
+      return arr[i];
+    }
+  }
+  return null;
+}
+
+interface CreationDetails {
+  __typename: string;
+  createdAt: string;
+  author: { login: string };
+}
+
+interface PullRequestReview extends CreationDetails {
+  __typename: 'PullRequestReview';
+  state:
+    | 'PENDING'
+    | 'COMMENTED'
+    | 'APPROVED'
+    | 'CHANGES_REQUESTED'
+    | 'DISMISSED';
+}
+
+interface ReviewRequestedEvent extends CreationDetails {
+  __typename: 'ReviewRequestedEvent';
+}
+
+interface GraphQLPRData {
+  mergedAt: string | null;
+  number: number;
+  merged: boolean;
+  createdAt: string;
+  author: { login: string };
+  timelineItems: {
+    nodes: Array<PullRequestReview | ReviewRequestedEvent>;
+  };
+}
+
+type TimelineItem = {
+  kind:
+    | 'PullRequestReview'
+    | 'ReviewRequestedEvent'
+    | 'ReopenedEvent'
+    | 'ReadyForReviewEvent';
+  datetime: DateTime;
+  login: string;
+  node: any;
+};
+
+export class PullRequest {
+  number: number;
+  mergedAt: DateTime | undefined;
```

Comment:
Any reason not to use optional property?

Reply:
👍 done